tuning/data/data_processors.py

```diff
@@ -283,16 +312,28 @@ def _process_dataset_configs(
                     column_names = raw_datasets[splitName].column_names
 
                     # remove __content__ from all processing
-                    if "__content__" in column_names:
+                    if not column_names and isinstance(
+                        raw_datasets, IterableDatasetDict
+                    ):
+                        logger.warning("Could not remove columns from IterableDataset")
+                    if column_names and "__content__" in column_names:
                         column_names.remove("__content__")
 
                     if "remove_columns" not in kwargs:
                         kwargs["remove_columns"] = None
                     if kwargs["remove_columns"] == "all":
                         kwargs["remove_columns"] = column_names
 
-                    if "num_proc" not in kwargs:
-                        kwargs["num_proc"] = os.cpu_count()
+                    if not isinstance(raw_datasets, IterableDatasetDict):
```

Comment:
minor nit: 
Could we just reverse this check for easier readablity?

```suggestion
                    if isinstance(raw_datasets, IterableDatasetDict):
                        if "num_proc" in kwargs:
                            del kwargs["num_proc"]
                            logger.warning(
                                "num_proc is not applicable for \
                                            IterableDatasets and has been removed."
                            )
                    else:
                        if "num_proc" not in kwargs:
                            kwargs["num_proc"] = os.cpu_count()
                            logger.info(f"setting num_proc to {os.cpu_count()}")
```